internal/repo/decompress.go

```diff
@@ -0,0 +1,153 @@
+package repo
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bytes"
+	"compress/bzip2"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"strings"
+
+	"github.com/ulikunitz/xz"
+)
+
+const (
+	exeNotFoundInArchive = "executable not found in archive"
+	cannotDecompressFile = "cannot decompress file"
+)
+
+var (
+	fileTypes = []struct {
+		ext        string
+		decompress func(src io.Reader, cmd, os, arch string) (io.Reader, error)
+	}{
+		{".zip", unzip},
+		{".tar.gz", untar},
+		{".tgz", untar},
+		{".gzip", gunzip},
+		{".gz", gunzip},
+		{".tar.xz", untarxz},
+		{".xz", unxz},
+		{".bz2", unbz2},
```

Comment:
Remove unnecessary support for other compression formats

Reply:
And the code for the function that follows

Reply:
Resolved